frontend/src/components/auth/Signup.tsx

```diff
@@ -13,12 +15,23 @@ const Signup = (): React.ReactElement => {
   const [email, setEmail] = useState("");
   const [password, setPassword] = useState("");
 
+  const location = useLocation();
+
+  // Extract query parameters from the URL
+  const params = new URLSearchParams(location.search);
+  const role = params.get("role");
+
+  if (role !== "facilitator") {
+    return <Redirect to="/welcome" />;
+  }
+
   const onSignupClick = async () => {
     const user: AuthenticatedUser = await authAPIClient.signup(
       firstName,
       lastName,
       email,
       password,
+      role.charAt(0).toUpperCase() + role.slice(1), // potential update in future
```

Comment:
yep! if you merge in the latest `main` branch into your branch (run `git pull origin main`), you'll be able to use Jeffrey's function for this - see `frontend/src/utils/StringUtils.ts`